cmd/subspace/net_setup.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/coreos/go-iptables/iptables"
+)
+
+var (
+	wireguardConfig struct {
+		gatewayIPv4 net.IP
+		gatewayIPv6 net.IP
+		networkIPv4 *net.IPNet
+		networkIPv6 *net.IPNet
+	}
+)
+
+func initWireguardConfig() error {
+	var err error
+	wireguardConfig.gatewayIPv4, wireguardConfig.networkIPv4, err = calcDefaultGateway(networkIPv4)
+	if err != nil {
+		return err
+	}
+	wireguardConfig.gatewayIPv6, wireguardConfig.networkIPv6, err = calcDefaultGateway(networkIPv6)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func configureWireguard() (string, error) {
+	var err error
+	iptable, err := iptables.New()
+	if err != nil {
+		return "", err
+	}
+	setupNAT := func(network, gateway string) error {
+		err = iptable.AppendUnique("nat", "POSTROUTING", "-s", network, "-j", "MASQUERADE")
+		if err != nil {
+			return err
+		}
+		err = iptable.AppendUnique("filter", "FORWARD", "-m", "state", "--state", "RELATED,ESTABLISHED", "-j", "ACCEPT")
+		if err != nil {
+			return err
+		}
+		err = iptable.AppendUnique("filter", "FORWARD", "-s", network, "-j", "ACCEPT")
+		if err != nil {
+			return err
+		}
+		// DNS Leak Protection
+		err = iptable.AppendUnique("nat", "OUTPUT", "-s", network, "-p", "udp", "--dport", "53", "-j", "DNAT", "--to", fmt.Sprintf("%s:53", gateway))
+		if err != nil {
+			return err
+		}
+		err = iptable.AppendUnique("nat", "OUTPUT", "-s", network, "-p", "tcp", "--dport", "53", "-j", "DNAT", "--to", fmt.Sprintf("%s:53", gateway))
+		if err != nil {
+			return err
+		}
+		return nil
+	}
```

Comment:
This can be a standalone function which will reduce the overall size of this one function

Reply:
Great work though on using `iptable` :D These contributions are very much appreciated!

Reply:
OK, I made a new standalone function, `setupNAT`.